splunk_connector.py

```diff
@@ -1341,6 +1330,26 @@ def _run_query(self, search_query, action_result, attach_result=False, kwargs_cr
         self.debug_print('Done run query')
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _wait_until_splunk_job_results_are_ready(self, action_result: phantom.ActionResult, job: splunk_client.Job, retry_limit: int) -> bool:
+        for attempt_count in range(0, retry_limit):
+            try:
+                #  Timing out the splunk job is required, because the job
+                #  could be stuck in permanent "QUEUED" state after the Splunk
+                #  stack has crashed.
+                max_waiting_time: float = time.time() + self.splunk_job_timeout
+                while not job.is_ready():
+                    if time.time() > max_waiting_time:
+                        return action_result.set_status(phantom.APP_ERROR, consts.SPLUNK_ERR_SPLUNK_JOB_HAS_TIMED_OUT)
+                    time.sleep(self.sleeptime_in_requests)
+                job.refresh()
+                break
+            except Exception as e:
+                if attempt_count == retry_limit - 1:
```

Comment:
can we make this a bit cleaner? Like for it to be attempt_count == retry_limit (change range appropriately for ex.)

Reply:
Sure, changed